auth.go

```diff
@@ -436,12 +461,23 @@ func (x *Central) GetTokenFromIdentityPassword(identity, password string) (*Toke
 // Create a new session. Returns a session key, which can be used in future to retrieve the token.
 // The internal session expiry is controlled with the member NewSessionExpiresAfter.
 // The session key is typically sent to the client as a cookie.
-func (x *Central) Login(username, password string) (sessionkey string, token *Token, err error) {
+func (x *Central) Login(username, password string, address string) (sessionkey string, token *Token, err error) {
 	user, authErr := x.authenticate(username, password)
 	if authErr != nil {
+
 		err = authErr
-		x.Stats.IncrementInvalidPasswords(x.Log)
 
+		if err == ErrInvalidPassword {
+			x.Stats.IncrementInvalidPasswords(x.Log)
+			x.Stats.IncrementInvalidPasswordHistory(x.Log, username, address)
+			if isAdmin, groupErr := x.CheckAdmin(user.UserId); groupErr == nil {
+				if int(x.Stats.UserLoginAttempts[username]) >= 3 && x.EnableAccountLocking && !isAdmin {
```

Comment:
3 is hardcoded. should use the configured limit.
Move the `x.EnableAccountLocking` check higher up, so that we don't even need to take the mutex lock if this functionality is not enabled.